prov/efa/src/efa_fabric.c

```diff
@@ -483,7 +483,7 @@ static int efa_get_gdr_support(char *device_name)
 	if (ret < 0)
 		goto out;
 
-	ret = atoi(&gdr_support);
+	ret = (int)(gdr_support - '0');
```

Comment:
This does not feel like the right fix; I'd much prefer we use `atoi()` or, better, `strtol()`.  We should not be reading the first character of the file (in the `fi_read_file()` call above and reading a string into a single character.  Much better would be to properly read the file into the string and then use the string parsing functions.

While you're in this function, please document inputs and outputs (especially given the other patch hacking on return values).

Reply:
Used a string for ```fi_read_file``` and ```strtol``` in new revision.